### PR TITLE
T3 changes for Redshift

### PR DESCRIPTION
**Pull request by azimnacho**

### CODE REVIEW

File: T3Viewer/db/views.py [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+
+
+class DBReportForm(forms.Form):
+    project = forms.ChoiceField(choices=[(x, x.capitalize()) for x in projects])
+    from_date = forms.DateTimeField(initial=datetime.now())
+    to_date = forms.DateTimeField(initial=datetime.now())
+
+    def clean_from_date(self):
+        from_date = self.cleaned_data.get('from_date', '')
+        if from_date.replace(tzinfo=None) > datetime.now():
```
Comment (jvilhuber) [inline T3Viewer/db/views.py:111]: There's no TZ check in the DBLoadForm. Should there be? Could these functions be combined?

File: T3Viewer/db/views.py [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
+    def clean_project(self):
+        project = self.cleaned_data.get('project', '')
+        if project not in projects:
+            self.add_error('project', 'Unknown Project')
+            raise ValidationError(_('Unknown Project: %(project)s'),
+                                  code='unknown',
+                                  params={'project': projects})
+        return project
+
+def nachotoken_required(view_func):
```
Comment (jvilhuber) [inline T3Viewer/db/views.py]: Should probably import this from where the other function for this is, so we can change both authenticates in one function when needed.

File: T3Viewer/db/views.py [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+    @wraps(view_func, assigned=available_attrs(view_func))
+    def _wrapped_view(request, *args, **kwargs):
+        return view_func(request, *args, **kwargs)
+        # if validate_session(request):
+        #     return view_func(request, *args, **kwargs)
+        # else:
+        #     return HttpResponseRedirect(settings.LOGIN_URL)
+    return _wrapped_view
+
+def nacho_cache(view_func):
```
Comment (jvilhuber) [inline T3Viewer/db/views.py]: Ditto.

File: T3Viewer/Bugfix/views.py [hunk 4]
```diff
@@ -355,11 +329,20 @@ def get_device_list_from_email(project, timestamp, email_address, span, event_cl
 
 def get_support_events(project, after, before, logger=None):
     conn = _aws_s3_connection(project)
-    bucket_name = projects_cfg.get(project, 'client_telemetry_bucket')
+    bucket_name = projects_cfg.get(project, 'client_t3_support_bucket')
```
Comment (azimnacho) [inline T3Viewer/Bugfix/views.py:311]: the project specifies 'dev/alpha/beta/prod'

File: T3Viewer/db/views.py (hunk 1 above)
Comment (azimnacho) [inline T3Viewer/db/views.py:111]: I didn't add those since we only deal with UTC in our logs

File: T3Viewer/db/views.py (hunk 2 above)
Comment (azimnacho) [inline T3Viewer/db/views.py]: yes. I need to clean this up. Currently this is not being used.

File: T3Viewer/Bugfix/views.py (hunk 4 above)
Comment (jvilhuber) [inline T3Viewer/Bugfix/views.py:311]: Oh that's a field in the config. Doh.

File: T3Viewer/templates/db_load.html [hunk 5]
```diff
@@ -0,0 +1,24 @@
+{% extends "base.html" %}
+{% block title %}Redshift Database Loader{% endblock %}
```
Comment (jvilhuber) [inline T3Viewer/templates/db_load.html:2]: I don't see a block named title in base.html. Should this be added or did I miss it? Or is this a django-ism that it knows to make this a title of some kind?

File: T3Viewer/Viewer/urls.py [hunk 6]
```diff
@@ -8,11 +8,17 @@
...
 urlpatterns = patterns('',
     url(r'^login/$', 'Bugfix.views.login'),
     url(r'^logout/$', 'Bugfix.views.logout', name='logout'),
-    url(r'^$', 'Bugfix.views.home'),
+    url(r'^$', 'Bugfix.views.index', name='index'),
+    url(r'^db/log-report-form$', 'db.views.db_log_report_form', name="log-report-form"),
+    url(r'^db/log-report/(?P<project>\w+)/(?P<from_date>%s)/(?P<to_date>%s)/$' % (timestamp_regex, timestamp_regex), 'db.views.db_log_report'),
+    url(r'^db/help$', 'db.views.db_help', name="db-help"),
+    url(r'^db/db-load$', 'db.views.db_load', name="db-load"),
+    url(r'^db/db-query$', 'db.views.db_query', name="db-query"),
```
Comment (jvilhuber) [inline T3Viewer/Viewer/urls.py:20]: Can report URL's still be shared? Or are they purely based on the POST data? I.e. if I find something weird in a report, can I just send you the URL and you'll get the same report?

**Comment by jvilhuber** [comment]
Looks fine. Let's try it out!

File: T3Viewer/templates/db_load.html (hunk 5 above)
Comment (azimnacho) [inline T3Viewer/templates/db_load.html:2]: it is there in the base.html. next to the 'Telemetry Viewer' :). Technically this should be called a subtitle.

File: T3Viewer/Viewer/urls.py (hunk 6 above)
Comment (azimnacho) [inline T3Viewer/Viewer/urls.py:20]: The report urls can be shared. eg . http://127.0.0.1:8081/db/log-report/dev/2015-06-20T16:00:44Z/2015-06-23T16:00:44Z/

**Comment by azimnacho** [comment]
cool. thanks, Jan.

File: T3Viewer/templates/db_load.html (hunk 5 above)
Comment (jvilhuber) [inline T3Viewer/templates/db_load.html:2]: Ah I see it now. Cool.

**Pull request merged**